File: #28 sistema_consultas/consulta_test.rb
# require_relative 'aaaa-s/nome'
require 'minitest/autorun'
require 'date'

class MedicoTest < Minitest::Test
  def setup
    @medico     = Medico.new('Dra. Satya Ananda')
    @paciente1  = Paciente.new('Élisson Michael')
    @paciente2  = Paciente.new('Brandon Nyorai')
  end

  def test_retorno_de_pacientes_de_uma_determinada_data
    @paciente1.agendar_consulta(@medico, Date.today)
    @paciente2.agendar_consulta(@medico, Date.today)
    assert_equal @medico.pacientes_do_dia(Date.today), [@paciente1, @paciente2]
    assert_equal @medico.pacientes_do_dia(Date.today + 1), []
  end

  def test_retorno_de_datas_em_que_atendeu_um_determinado_paciente
    @paciente1.agendar_consulta(@medico, Date.new(2018, 4, 3))
    assert_equal @medico.datas_de_atendimento_do(@paciente1), ['03/04/2018']
    assert_equal @medico.datas_de_atendimento_do(@paciente2), []
  end
end

File: #28 sistema_consultas/ConsultaTest.java
import static org.junit.Assert.assertEquals;
import java.util.Arrays;
import java.util.Date;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class MedicoTest {
	Medico medico;
	Paciente paciente1;
	Paciente paciente2;
	Date data;

	@BeforeEach
	void setUp() throws Exception {
		medico = new Medico("Dra. Satya Ananda");
		paciente1 = new Paciente("Élisson Michael");
		paciente2 = new Paciente("Brandon Nyorai");
		data = new Date(2018, 3, 16);
	}

	@Test
	void testRetornodePacientesdeumaDeterminadaData() {
		paciente1.agendarConsulta(medico, data);
		paciente2.agendarConsulta(medico, data);
		assertEquals(Arrays.asList(paciente1, paciente2), medico.pacientesDoDia(data));
		assertEquals(Arrays.asList(), medico.pacientesDoDia(new Date(2018, 3, 17)));
	}

	@Test
	void testRetornoDeDatasEmQueAtendeuUmDeterminadoPaciente() {
		paciente1.agendarConsulta(medico, data);
		assertEquals(Arrays.asList("16/04/18"), medico.datasDeAtendimentoDo(paciente1));
		assertEquals(Arrays.asList(), medico.datasDeAtendimentoDo(paciente2));
	}
}
